src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -70,6 +70,17 @@ public static bool InStandbyMode
             }
         }
 
+        public static string InstanceId
+        {
+            get
+            {
+                string id = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")
```

Comment:
I can refactor so that _some_ of the logic would be shared, but `Environment.MachineName` would not work for what we require, as in the case of a proposed lease ID, it has to be a value that can be parsed into a Guid.

As implemented, the `InstanceId` will return a 32-character hex value (derived from the instance ID if running on Azure or from the `Guid` we create there). 

Thoughts?